From 70e6aacc9c0f4843b4a04b66893cd9bbdd0e0e78 Mon Sep 17 00:00:00 2001
From: Xie xiaobo <X.Xie@freescale.com>
Date: Mon, 4 May 2009 15:52:57 +0800
Subject: [PATCH] ENGR00112162-1 MX25: Fix remove camera module failure case

Camera module on MX25 cannot be removed successfully. This patch is to
change sensor gpio to fix.

Signed-off-by: Guoqing Jiang <b22156@freescale.com>
---
 arch/arm/mach-mx25/mx25_3stack_gpio.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx25/mx25_3stack_gpio.c b/arch/arm/mach-mx25/mx25_3stack_gpio.c
index eee3432..55f5a79 100644
--- a/arch/arm/mach-mx25/mx25_3stack_gpio.c
+++ b/arch/arm/mach-mx25/mx25_3stack_gpio.c
@@ -984,7 +984,7 @@ void gpio_sensor_active(void)
 	mxc_request_iomux(MX25_PIN_CSI_PIXCLK, MUX_CONFIG_FUNC);
 	mxc_request_iomux(MX25_PIN_CSI_VSYNC, MUX_CONFIG_FUNC);
 	mxc_request_iomux(MX25_PIN_A19, MUX_CONFIG_ALT5); /*CSI_PWDN*/
-	mxc_request_iomux(MX25_PIN_A20, MUX_CONFIG_ALT5);
+	mxc_request_iomux(MX25_PIN_A20, MUX_CONFIG_ALT5); /*CMOS_RST*/
 
 	mxc_set_gpio_direction(MX25_PIN_A19, 0); /*CSI_PWDN*/
 	mxc_set_gpio_dataout(MX25_PIN_A19, 0);
@@ -1031,6 +1031,8 @@ void gpio_sensor_inactive(void)
 	mxc_request_gpio(MX25_PIN_CSI_PIXCLK);
 	mxc_request_gpio(MX25_PIN_CSI_VSYNC);
 
+	mxc_free_iomux(MX25_PIN_A19, MUX_CONFIG_GPIO);
+	mxc_free_iomux(MX25_PIN_A20, MUX_CONFIG_GPIO);
 	mxc_free_iomux(MX25_PIN_CSI_D2, MUX_CONFIG_GPIO);
 	mxc_free_iomux(MX25_PIN_CSI_D3, MUX_CONFIG_GPIO);
 	mxc_free_iomux(MX25_PIN_CSI_D4, MUX_CONFIG_GPIO);
-- 
1.5.4.4

